app/core/models/ocelot/package.scala

```diff
@@ -47,6 +52,13 @@ package object ocelot {
   def pageLinkIds(phrases: Seq[Phrase]): List[String] = phrases.flatMap(phrase => pageLinkIds(phrase.english)).toList
   def labelReferences(str: String): List[String] = plSingleGroupCaptures(labelRefRegex, str)
   def labelReference(str: String): Option[String] = plSingleGroupCaptures(labelRefRegex, str).headOption
+  def listLength(listName: String, labels: Labels): Option[String] = labels.valueAsList(listName).fold[Option[String]](None){l => Some(l.length.toString)}
+  def labelScalarMatch(m: Regex.Match, labels: Labels, lbl: String => Option[String]): Option[String] =
+    Option(m.group(1)).fold[Option[String]]{
+      Option(m.group(4)).fold[Option[String]](None)(list => listLength(list, labels))
+    }{label => lbl(label)}
+  def labelScalarValue(str: String)(implicit labels: Labels): Option[String] =
```

Comment:
Again, I wonder if this method should be in in Labels/LabelCacheImpl. It is the class with the primary responsibility for storing labels and giving access to their values.

Reply:
as above but more so, this fn is primarily applying a regex to a string to detect use of a placeholder unrelated to the essential Labels interface

Reply:
As above, but it is probably more relevant for this method.